### PR TITLE
Slash fixes

### PR DESCRIPTION
**Pull request by alentoghostflame**
SlashOption's choice param now takes an iterable.
rollout_all_guilds kwarg for Client now determines if all Guilds are rollout'd to instead of just the ones specified by the guild_ids param in commands. This is a fix towards the cloudflare ban.
handle_slash_argument is now async, and will auto-fetch user/member if they aren't found in the cache.
Added basic cog reloading support, fix application commands not actually getting removed.
Guild now has its own rollout command.

## Summary

Adds a fix towards #310, but the only real fix for that is fixing http.
Contains the same exact fix as #313, sorry.
When a member/user can't be gotten from the cache (hi intents!), slash commands will now attempt to fetch it.
Adds a variety of fixes and a few helpers for slash stuff.

## Checklist

- [X] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [X] This PR fixes an issue.
- [X] This PR adds something new (e.g. new method or parameters).
- [X] This PR is a breaking change (e.g. methods or parameters removed/renamed) [for some inner slash stuff]
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by VincentRPS** [comment]
@alentoghostflame client.add_cog exists but there is no way to actually make a cog without commands.Bot

**Comment by VincentRPS** [comment]
@alentoghostflame client.add_cog exists but there is no way to actually make a cog without commands.Bot

**Comment by alentoghostflame** [comment]
> @alentoghostflame client.add_cog exists but there is no way to actually make a cog without commands.Bot

Two things about this:
1: ClientCog exists.
2: Discussions about bringing cogs into Client have been made, so it will likely be relevant for the future.

**Comment by VincentRPS** [comment]
On number 2 why should they be moved? Cogs aren’t a discord feature in anyway so we shouldn’t treat it like one

**Comment by Skelmis** [comment]
> On number 2 why should they be moved? Cogs aren’t a discord feature in anyway so we shouldn’t treat it like one, since every feature in nextcord/ is treated as a discord feature and every thing nextcord/ext an extension

I believe it comes under 'implementation details', you can't tell me that if you were making a bot using just ``discord.Client`` you *wouldn't* want the ability to separate your code into multiple files to make it overall better?

Tbh I'm surprised Cogs weren't a thing in client from the start

**Comment by alentoghostflame** [comment]
Cogs are the only way currently to easily break a project up into multiple files when using Slash or Listeners/Events.

**Comment by VincentRPS** [comment]
If you want to use cogs just use commands.Bot, There’s no downside to it

**Comment by Skelmis** [comment]
As a fork we have been given a chance to *improve* useability, theres nothing wrong with porting the ability to code a bot in multiple files. Your argument is just we don't do it now, why bother

File: nextcord/guild.py [hunk 1]
```diff
@@ -2994,6 +2994,44 @@ async def deploy_application_commands(
...
+        """|coro|
+        Rolls out application commands to the guild, associating, deleting, updating, and/or newly
+        registering as needed.
+
+        Parameters
+        ----------
+        associate_known: :class:`bool`
+            Whether commands on Discord that match a locally added command should be associated with each other.
+            Defaults to ``True``
+        delete_unknown
```
Comment (ooliver1) [inline nextcord/guild.py:3013]: will these get any docs

**Review by vcokltfre — APPROVED** [review]

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**